github/repos_contents.go

```diff
@@ -267,6 +267,12 @@ func (s *RepositoriesService) GetArchiveLink(owner, repo string, archiveformat a
 	} else {
 		resp, err = s.client.client.Transport.RoundTrip(req)
 	}
+	defer func() {
+		// Make sure to close the response body if we have one
+		if resp != nil {
+			resp.Body.Close()
+		}
+	}()
 	if err != nil || resp.StatusCode != http.StatusFound {
 		return nil, newResponse(resp), err
 	}
```

Comment:
I thought about the same thing myself as well, but thought to change as little as possible... 

But I fully agree so will update the PR in a moment.